lib/ex_format.ex

```diff
@@ -1002,15 +1021,27 @@ defmodule ExFormat do
     :binary.part(char, 1, byte_size(char) - 2)
   end
 
-  defp format_integer_literal({:__block__, meta, [int]} = ast, fun) do
+  defp format_literal({:__block__, meta, [literal]} = ast, fun) do
     expr =
       case meta[:format] do
-        :char -> "?" <> codepoint_to_string(int)
-        :binary -> "0b" <> Integer.to_string(int, 2)
-        :octal -> "0o" <> Integer.to_string(int, 8)
-        :hexadecimal -> "0x" <> Integer.to_string(int, 16)
-        :decimal -> Integer.to_string(int) |> underscores_in_decimal()
-        _ -> Integer.to_string(int)
+        :char ->
+          "?" <> codepoint_to_string(literal)
+        :binary ->
+          "0b" <> Integer.to_string(literal, 2)
+        :octal ->
+          "0o" <> Integer.to_string(literal, 8)
+        :hexadecimal ->
+          "0x" <> Integer.to_string(literal, 16)
+        :decimal ->
+          Integer.to_string(literal) |> underscores_in_decimal()
+        :bin_heredoc ->
+          "\"\"\"\n" <> literal <> "\"\"\""
+        :list_heredoc when is_binary(literal) ->
+          "'''\n" <> literal <> "'''"
```

Comment:
It happens when there is an interpolated heredoc charlist:

```
'''
#{:hello}
'''
```

The AST would be:

```
 {{:., [line: 1], [String, :to_charlist]}, [line: 1],
  [{:<<>>, [format: :list_heredoc, line: 1],
    [{:::, [line: 2],
      [{{:., [line: 2], [Kernel, :to_string]}, [line: 2], [:hello]},
       {:binary, [line: 2], nil}]}, "\n"]}]}
```